How to check whether the following is safe:  
1 3 6 7 9

Approach:  
construct vector with index, put in complex plane, use norm and angle.

input : 1, 3, 6, 7, 9  
index:  1, 2, 3, 4, 5  

So we have vectors:  
- (1, 1)
- (2, 3)
- (3, 6)
- (4, 7)
- (5, 9)  

So we have complex numbers:  
- 1 + 1i
- 2 + 3i
- 3 + 6i
- 4 + 7i
- 5 + 9i 



In [9]:
import cmath

cnum1 = complex(1, 1)
cnum2 = complex(2, 3)
cnum3 = complex(3, 6)
cnum4 = complex(4, 7)
cnum5 = complex(5, 9) 

print(cnum1)
print(cnum2)
print(cnum3)
print(cnum4)
print(cnum5)




(1+1j)
(2+3j)
(3+6j)
(4+7j)
(5+9j)


We are going to study the subtraction of each two adjacent complex number:  
(=we are going to study the delta of each step)

In [10]:
cnum2_minus_1 = cnum2 - cnum1
cnum3_minus_2 = cnum3 - cnum2
cnum4_minus_3 = cnum4 - cnum3
cnum5_minus_4 = cnum5 - cnum4

print(cnum2_minus_1)
print(cnum3_minus_2)
print(cnum4_minus_3)
print(cnum5_minus_4)

(1+2j)
(1+3j)
(1+1j)
(1+2j)


now we have the subject to study. They are complex number, vector, ...whatever.  

"Two adjacent levels differ by at least one" --> the delta complex number is 1 + 1 i  
We can see that for this complex, its angle (or phase?phi?psi?theta? whatever..) is:  

In [14]:
cmath.polar(complex(1,1))[1]

0.7853981633974483

"... at most three":  
the delta complex number is 1 + 3 j,  
and its angle:  

In [15]:
cmath.polar(complex(1,3))[1]

1.2490457723982544

Thus, for each delta, the angle should be between 0.7853981633974483 and 1.2490457723982544.  
And the sign (+, -) of angle tells whether delta increase or decrease